app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,43 @@
+class VotesController < ApplicationController
+  def upvote
+    @vote = Vote.find_or_initialize_by(user_id: current_user.id, review_id: params[:id])
+    if @vote.new_record? == true
+      @vote.entry = 1
+      @vote.save
+      flash[:alert] = "Your vote has been saved."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    elsif @vote.new_record? == false && @vote.entry == -1
+      @vote.entry = 1
+      @vote.save
+      flash[:alert] = "Your vote has been updated."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    else
+      flash[:alert] = "You have already up-voted for this review."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    end
+  end
+
+  def downvote
+    @vote = Vote.find_or_initialize_by(user_id: current_user.id, review_id: params[:id])
+    if @vote.new_record? == true
+      @vote.entry = -1
+      @vote.save
+      flash[:alert] = "Your vote has been saved."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    elsif @vote.new_record? == false && @vote.entry == 1
+      @vote.entry = -1
+      @vote.save
+      flash[:alert] = "Your vote has been updated."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    else
+      flash[:alert] = "You have already down-voted for this review."
+      redirect_to playlist_path(Review.find(params[:id]).playlist.id)
+    end
+  end
```

Comment:
Stick to the default resource actions (index, show, new, create, edit, update, destroy) unless you have a good reason not to. In this case, `upvote` and `downvote` are really special cases of `create` or `update`. You can specify whether it is an upvote or downvote by passing in a value through the form (as a hidden field): a value of 1 is an upvote, -1 is a downvote.